src/Servers/Kestrel/Core/src/Internal/SniOptionsSelector.cs

```diff
@@ -84,7 +84,8 @@ internal class SniOptionsSelector
 
                 if (clientCertificateMode != ClientCertificateMode.NoCertificate)
                 {
-                    sslOptions.ClientCertificateRequired = true;
+                    sslOptions.ClientCertificateRequired = clientCertificateMode == ClientCertificateMode.AllowCertificate
```

Comment:
Why is 'allow' setting required?

Reply:
ClientCertificateRequired is misleading, all it really means is should it ask the client for a cert or not.

> Gets or sets a value that specifies whether the client is asked for a certificate for authentication. Note that this is only a request -- if no certificate is provided, the server still accepts the connection request.

https://docs.microsoft.com/en-us/dotnet/api/system.net.security.sslserverauthenticationoptions.clientcertificaterequired?view=net-5.0#System_Net_Security_SslServerAuthenticationOptions_ClientCertificateRequired

Reply:
Figured it was something weird like that